### PR TITLE
✨[amp-story-player] Adds tap-to-next/previous story

### LINKED ISSUE #26697: [amp-story-player] Add tap-to-next story.
Whenever a user is finished reading a story, the next tap should take the user to the next story.

### PR DESCRIPTION
**Pull request by Enriqe**
closes #26697

Tracker #26308

### CODE REVIEW

File: src/amp-story-player.js [hunk 1]
```diff
@@ -171,6 +171,14 @@ export class AmpStoryPlayer {
...
         messaging => {
           messaging.setDefaultHandler(() => Promise.resolve());
+
+          messaging.registerHandler('selectDocument', (event, data) => {
+            if (data.next) {
+              this.next_();
+            } else if (data.previous) {
+              this.previous_();
+            }
+          });
```
Comment (gmajoulet) [inline src/amp-story-player.js]: Nit: Let's start extracting the handlers to different methods, as they might grow a lot (`onSelectDocument_()`)

File: src/amp-story-player.js [hunk 2]
```diff
@@ -256,12 +263,20 @@ export class AmpStoryPlayer {
       previousStory[IFRAME_IDX],
       VisibilityState.INACTIVE
     );
+    this.iframes_[previousStory[IFRAME_IDX]].setAttribute(
+      'i-amphtml-story-position',
```
Comment (gmajoulet) [inline src/amp-story-player.js]: Plz either define a top level constant, or a private helper

**Comment by Enriqe** [comment]
Added some basic unit tests, PTAL :)

File: test/unit/test-amp-story-player.js [hunk 3]
```diff
@@ -39,6 +43,22 @@ describes.realWin('AmpStoryPlayer', {amp: false}, env => {
...
   beforeEach(() => {
     win = env.win;
+
+    fakeMessaging = {
+      setDefaultHandler: () => {},
+      registerHandler: (event, handler) => {
+        fireHandler = handler;
+      },
+    };
+    fakeMessagingPromise = new Promise(resolve => {
```
Comment (gmajoulet) [inline test/unit/test-amp-story-player.js]: `*.resolves(foo)` is a shorthand for `*.returns(Promise.resolve(foo))`
You should be able to directly call `.resolves(fakeMessaging)` without this extra variable :)

File: test/unit/test-amp-story-player.js [hunk 4]
```diff
@@ -39,6 +43,22 @@ describes.realWin('AmpStoryPlayer', {amp: false}, env => {
 
   beforeEach(() => {
     win = env.win;
+
+    fakeMessaging = {
```
Comment (gmajoulet) [inline test/unit/test-amp-story-player.js]: Nit: `const` (doesn't need to be a top level `let`)

File: test/unit/test-amp-story-player.js [hunk 5]
```diff
@@ -138,4 +158,36 @@ describes.realWin('AmpStoryPlayer', {amp: false}, env => {
       expect(stories[3][IFRAME_IDX]).to.eql(undefined);
     }
   );
+
+  it('should register handlers at build time', async () => {
+    buildStoryPlayer();
+    const player = new AmpStoryPlayer(win, playerEl);
+
+    env.sandbox
```
Comment (gmajoulet) [inline test/unit/test-amp-story-player.js]: If you think these lines will be repeated in every single test, should we define a helper that creates the player + stubs `initializeHandshake_`?

File: test/unit/test-amp-story-player.js [hunk 6]
```diff
@@ -138,4 +158,36 @@ describes.realWin('AmpStoryPlayer', {amp: false}, env => {
...
+      .stub(player, 'initializeHandshake_')
+      .resolves(fakeMessagingPromise);
+
+    await player.buildCallback();
+    player.layoutCallback();
+
+    const fakeData = {next: true};
+    fireHandler('selectDocument', fakeData);
+
+    expect(nextSpy).to.be.called;
```
Comment (gmajoulet) [inline test/unit/test-amp-story-player.js]: Let's assert that the next story is actually selected instead? From your previous tests it looks straightforward, and would give a much better coverage

**Pull request merged**